src/components/PDFView/PDFPasswordForm.js

```diff
@@ -45,133 +41,119 @@ const defaultProps = {
-        const containerStyle = this.props.isSmallScreenWidth ? [styles.flex1, styles.w100] : styles.pdfPasswordForm.wideScreenWidth;
-
-        return (
-            <>
-                {this.state.shouldShowForm ? (
-                    <ScrollView
-                        keyboardShouldPersistTaps="handled"
-                        style={containerStyle}
-                        contentContainerStyle={styles.p5}
-                    >
-                        <View style={styles.mb4}>
-                            <Text>{this.props.translate('attachmentView.pdfPasswordForm.formLabel')}</Text>
-                        </View>
-                        <TextInput
-                            ref={(el) => (this.textInputRef = el)}
-                            label={this.props.translate('common.password')}
-                            accessibilityLabel={this.props.translate('common.password')}
-                            accessibilityRole={CONST.ACCESSIBILITY_ROLE.TEXT}
-                            /**
-                             * This is a workaround to bypass Safari's autofill odd behaviour.
-                             * This tricks the browser not to fill the username somewhere else and still fill the password correctly.
-                             */
-                            autoComplete={Browser.getBrowser() === CONST.BROWSER.SAFARI ? 'username' : 'off'}
-                            autoCorrect={false}
-                            textContentType="password"
-                            onChangeText={this.updatePassword}
-                            returnKeyType="done"
-                            onSubmitEditing={this.submitPassword}
-                            errorText={errorText}
-                            onFocus={() => this.props.onPasswordFieldFocused(true)}
-                            onBlur={this.validateAndNotifyPasswordBlur}
-                            autoFocus
-                            shouldDelayFocus={shouldDelayFocus}
-                            secureTextEntry
-                        />
-                        <Button
-                            text={this.props.translate('common.confirm')}
-                            onPress={this.submitPassword}
-                            style={styles.mt4}
-                            isLoading={this.props.shouldShowLoadingIndicator}
-                            pressOnEnter
-                        />
-                    </ScrollView>
-                ) : (
-                    <View style={[styles.flex1, styles.justifyContentCenter]}>
-                        <PDFInfoMessage onShowForm={this.showForm} />
+    };
+
+    const submitPassword = () => {
+        if (!validate()) {
+            return;
+        }
+        onSubmit(password);
+    };
+
+    const validateAndNotifyPasswordBlur = () => {
+        validate();
+        onPasswordFieldFocused(false);
+    };
+
+    const showForm = () => setShouldShowForm(true);
```

Comment:
What I meant is to pass the function directly to `PDFInfoMessage` without defining it.

```js
<PDFInfoMessage onShowForm={() => setShouldShowForm(true)} />
```